### PR TITLE
KNIME-2089: Introduce sanitization for intermediate products (#156)

### PR DESCRIPTION
**Pull request by manuelschwarze**
Includes version bump to 4.12.0 (for KNIME <5.3), to be cherry-picked later into master for KNIME 5.3+.

### CODE REVIEW

**Comment by manuelschwarze** [comment]
Tested successfully by me on Linux, Windows and Mac (Intel). Reviewed and approved already by Paolo (in our Novartis BitBucket). From my side ready to merge.

File: org.rdkit.knime.nodes/src/org/rdkit/knime/nodes/chemicaltransformation/RDKitChemicalTransformationNodeModel.java [hunk 1]
```diff
@@ -379,7 +404,7 @@ public DataCell[] process(final InputDataInfo[] arrInputDataInfo, final DataRow
 									// Increase the cycle number as we had a valid reaction
 									iCycle++;
 
-									// Instead of sanitization just update the property cache
+									// Update the property cache (might not be necessary since we sanitized already)
```
Comment (greglandrum) [inline org.rdkit.knime.nodes/src/org/rdkit/knime/nodes/chemicaltransformation/RDKitChemicalTransformationNodeModel.java]: This is definitely not required and can be removed.

**Review by greglandrum — CHANGES_REQUESTED** [review]
I think it's good to remove the redundant call to `updatePropertyCache()`

File: org.rdkit.knime.nodes/src/org/rdkit/knime/nodes/chemicaltransformation/RDKitChemicalTransformationNodeModel.java (hunk 1 above)
Comment (manuelschwarze) [reply org.rdkit.knime.nodes/src/org/rdkit/knime/nodes/chemicaltransformation/RDKitChemicalTransformationNodeModel.java]: Done. Tests are still working the same (tested only on Linux).

**Pull request merged**